mathoxide-lib/src/array.rs

```diff
@@ -32,32 +31,44 @@ where
     StorageType: Storage<Stored = T>,
     ViewType: ArrayView,
 {
-    pub fn offset(&self) -> usize {
+    pub fn storage_offset(&self) -> usize {
         self.view.offset()
     }
 
-    pub fn ndims(&self) -> usize {
-        self.view.ndims()
+    pub fn shape(&self) -> &[usize] {
+        self.view.shape()
     }
 
-    pub fn size(&self) -> usize {
-        self.view.size()
+    pub fn stride(&self) -> &[usize] {
+        self.view.stride()
     }
 
-    pub fn shape(&self) -> &[usize] {
-        self.view.shape()
+    pub fn ndim(&self) -> usize {
+        self.view.ndim()
+    }
+
+    pub fn numel(&self) -> usize {
+        self.view.numel()
+    }
+
+    pub fn is_contiguous(&self) -> bool {
+        self.view.is_contiguous()
+    }
+
+    pub fn storage_size(&self) -> usize {
+        self.storage.storage_get().iter().len()
```

Comment:
If i understand correctly the `len` method will go over all the iterator items to compute the length. Would ti make sense to cache the result? Doesn't have to be in this PR.

Reply:
I was also not sure about this one. `count()` definitely goes over the iterator items. `len()` is unclear to me. Shouldn't there be an easier way to directly getting length without going through the iterator API? Maybe add something to Storage?